### ISSUE TITLE
[TPL2] Second port of the execution engine

### ISSUE THREAD

**Issue by lmwnshn** [cmu-db/noisepage#914]
TLDR status update: 
- 2020-06-09: think we have the (talks to terrier / wan) and (tpl stuff / tanuj) pieces built out, we need to connect them now.
- 2020-05-22: porting over the supporting code, haven't touched the meat of it.

---

The plan is for @amlatyrngom @lmwnshn @tanujnay112 to port over the new version of @pmenon's execution engine. This will be done in pieces as far as is possible, i.e., we will try to go the "add new functionality without using it" route and try to keep the `terrier`, `tpl`, and `check-tpl` targets working at all times.

We are currently farming out work by the sub-directory in Prashanth's src/ branch, namely,

- [ ] ast <-- @lmwnshn #915
- [ ] common
- [ ] logging
- [ ] parsing
- [ ] sema <-- @tanujnay112 PR to be opened next
- [ ] sql <-- runtime types and values (#917), more to follow...
- [ ] util
- [ ] vm <-- formerly @lmwnshn, currently on hold until I understand what is in sql/

**Comment by lmwnshn** [comment]
Wow I did not keep this up to date. TPL2 merged. Closing.
